app/Actions/Sport/GetCountryMedalsAction.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Actions\Sport;
+
+use App\Models\Country;
+use Closure;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+
+class GetCountryMedalsAction
+{
+    public function handle(): Collection
+    {
+        return Country::query()
+            ->withCount([
+                'medals as countries_gold_count' => $this->withMedalsCountCallback('gold'),
```

Comment:
@babul28 not sure if a separate protected function is shorter than same callback function inline. Now, my eyes (as a new developer) needed to jump up-down to understand what that separate method does, so main query is not that readable.

Actually, probably will shoot a separate video about it, and discuss with the audience :)

Reply:
thank you for your advice, i will wait for it 👍